ompi/mca/sharedfp/sm/sharedfp_sm.c

```diff
@@ -94,6 +94,37 @@ struct mca_sharedfp_base_module_1_0_0_t * mca_sharedfp_sm_component_file_query(o
             return NULL;
         }
     }
+
+
+    /* Check that we can actually open the required file */
+    char *filename_basename = basename((char*)fh->f_filename);
+
+    /* format is "%s/%s_cid-%d-%d.sm", see below */
+    int sm_filename_length = strlen(ompi_process_info.job_session_dir) + 1 + strlen(filename_basename) + 5 + (3*sizeof(uint32_t)+1) + 4;
```

Comment:
Maybe just use asprintf()?

Reply:
you are right, thanks! This simplifies the code a bit.